### DISCUSSION TITLE
Parsing CString array with given total length

### DISCUSSION THREAD

**Discussion by dougsm** [MatrixEditor/caterpillar#14] [Q&A] [answered]
Hi, thanks for the great library!  I'm struggling with understanding how to handle the following case:

I need to parse an array of null-terminated `CString`s.   I don't know exactly how many strings are present, but I know how many bytes total the array of strings take up.  The array logic seems to only allow for the number of objects not bytes.  I can't use greedy parsing because the following data can sometimes look "string like".   Ideally what I'd like to achieve is "do greedy parsing, but only on the next `string_array_length_bytes` bytes.   

example:

``` python

@struct
class ExampleCStringArray:
    string_array_length_bytes: uint8
    string_array: CString[???]  # How to parse this array?  
    some_other_data_that_may_look_like_a_string: ...
```

Thanks!

**Comment by MatrixEditor** [comment, marked as answer]
Hi @dougsm, if I understand your case correctly you want something similar to this: [Chaining](https://matrixeditor.github.io/caterpillar/tutorial/advanced.html#chaining)

The structure would then look like this:
```
@struct
class Format:
    length: uint8
    values: Bytes(this.length) & CString[...]
    ...
```

This code will create a new temporary parsing stream for the amount of N next bytes and will apply greedy parsing only within that new stream.

**Reply by dougsm** [reply]
Thankyou, that's exactly what I needed 🙏
